modules/backend/controllers/Files.php

```diff
@@ -51,14 +52,54 @@ public function thumb($code = null, $width = 100, $height = 100, $mode = 'auto',
         return Response::make(View::make('backend::404'), 404);
     }
 
+    /**
+     * Attempt to return a redirect to a temporary URL to the asset instead of streaming the asset - if supported
+     *
+     * @param System|Models\File $file
+     * @param string|null $path Optional, defaults to the getDiskPath() of the file
+     * @return string|null
+     */
+    protected static function getTemporaryUrl($file, $path = null)
+    {
+        // Get the disk and adapter used
+        $url = null;
+        $disk = $file->getDisk();
+        $adapter = $disk->getAdapter();
+        if (is_a($adapter, 'League\Flysystem\Cached\CachedAdapter')) {
```

Comment:
@LukeTowers Any thoughts on switching these calls to `instanceof`? I find they're more readable, but it's a matter of preference.

Reply:
6 of one, half a dozen of another. Doesn't really matter to me, `instanceof` is used elsewhere in October and Laravel so that's a valid reason to do that. I was mostly just experimenting with `is_a()` because I recently rediscovered it today.